### ISSUE TITLE
[BUG] Packaging plugins through "grafana-toolkit" has problems

### ISSUE THREAD

**Issue by amotl** [grafana/grafana#20071]
Hi there,

similar to #13996, installing a plugin directly from a .zip file fails. We used the link from [1] and invoked an installation command like
```bash
grafana-cli --pluginUrl https://64-162061833-gh.circle-artifacts.com/0/home/circleci/plugin/ci/packages/grafana-worldmap-panel-0.3.0.zip plugins install grafana-worldmap-panel
```

What we are getting from this is a plugin directory looking like 
![image](https://user-images.githubusercontent.com/453543/65824634-89567200-e26c-11e9-912e-f5f260651d17.png)
where all the main files would be dropped into the `plugins` folder itself while all assets (`.html`, `.json`, `.css` and `.png` files) would be inside the `grafana-worldmap-panel` subfolder.

Invoking this on a locally built plugin like
```
grafana-cli --pluginUrl /Users/amo/dev/hiveeyes/sources/grafana-worldmap-panel/ci/packages/grafana-worldmap-panel-0.5.0.zip plugins install grafana-worldmap-panel 0.5.0
```
gives the same result. It looks like the designated `<plugin id>` and `<plugin version (optional)>` parameters `grafana-worldmap-panel 0.5.0` are not honored at all or improperly.

May we humbly ask if you could take a look what might be going wrong with the plugin installation process here? We copied this from [2] and humbly would like to ping @ryantxu or @dprokop regarding this issue.

Thanks already for taking the time to look into this issue!

With kind regards,
Andreas.

P.S.: We are running Grafana 6.4.2 on macOS installed through Homebrew.

[1] https://circleci.com/gh/panodata/grafana-worldmap-panel/64#artifacts/containers/0
[2] https://github.com/panodata/grafana-worldmap-panel/issues/15

**Comment by aocenas** [comment]
@amotl thanks for reporting this. How does the zip archive looks like when unzipped manually? Does it contain all files directly or is there a directory and then files?

@marefr Can you take a look at this? I think we assume the zip file contains a directory and inside that directory all the files so that when unzipped it is already in a directory.

**Comment by amotl** [comment]
Dear @aocenas,

thanks for taking the time to look into this. You are right, the zip archive just contains the files without any directory.

We have been using the `grafana-toolkit` task like
```
npx grafana-toolkit plugin:ci-package
```
on the repository [1] as proposed by @ryantxu (thanks again!).

We successfully resolved the dependencies on `"@grafana/toolkit": "next"` through `yarn install`.

So, we may conclude the packing bundling process of `grafana-toolkit` does not honor this important detail appropriately yet?

With kind regards,
Andreas.

[1] https://github.com/panodata/grafana-worldmap-panel

**Comment by marefr** [comment]
This might be a bug in grafana-cli. Will verify using older Grafana versions.

**Comment by amotl** [comment]
Dear @marefr,

thank you so much. The bug might either be in `grafana-cli` or inside the package bundling routines of the new `grafana-toolkit`, depending on how graceful `grafana-cli plugins install` should actually be with respect to the structure of the zip archive (with or without a directory inside).

We also reopened https://github.com/panodata/grafana-worldmap-panel/issues/8 in order to appropriately track both perspectives accordingly.

With kind regards,
Andreas.

**Comment by marefr** [comment]
Tried to install plugin using Grafana v5.3.0 in docker

```bash
grafana-cli --pluginUrl https://64-162061833-gh.circle-artifacts.com/0/home/circleci/plugin/ci/packages/grafana-worldmap-panel-0.3.0.zip plugins install grafana-worldmap-panel
```

and it results in the same installer file/directory structure in the plugins install directory.

@dprokop can you have a look to see whether there's a problem with @grafana/toolkit and/or a configuration error?

**Comment by marefr** [comment]
@dprokop I guess we can support both. Please open an issue for supporting that. Thanks

**Comment by dprokop** [comment]
@amotl I have created a PR that should address this issue on grafana/toolkit side (#20123). After @ryantxu reviewed it we will merge and you will be able to test it again. I suggest to use `canary` npm tag rather than `next` - it's a version that's not released yet and is built from master.

**Comment by amotl** [comment]
Thanks a bunch!

**Comment by amotl** [comment]
There was a minor glitch with #20123 but we have been able to fix it up through #20139 and humbly ask for a review.

**Comment by amotl** [comment]
After the fixup coming from #20139, we are able to confirm installation through e.g.
```
grafana-cli --pluginUrl ci/packages/grafana-worldmap-panel-0.5.0.zip plugins install grafana-worldmap-panel
```
works like a charm.

Thanks again for supporting me with this matter.

---

As a quick note: At least when using the `--pluginUrl` option, the folder name which gets materialized within the Grafana `plugins` folder at `/usr/local/var/lib/grafana/plugins` will be determined through the `<plugin id>` argument.

So if you say `foobar` there, the contents will get extracted into a folder named `foobar`. The optional `<plugin version>` argument will apparently not be used for anything.

Personally, I don't care about that detail, but it might bite others as I believe Grafana will pick up plugins through their `plugin.json` file, so typos on the installation process might bite some people when accidentally having multiple same plugins in different folders.

---

```
offgrid:~ amo$ grafana-cli plugins --help
NAME:
   Grafana cli plugins - Manage plugins for grafana

USAGE:
   Grafana cli plugins command [command options] [arguments...]

COMMANDS:
     install                  install <plugin id> <plugin version (optional)>
```
